## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [12]:
!pip install statsmodels

In [4]:
import pandas as pd
import numpy as np
from statsmodels.stats.weightstats import ztest

In [8]:
df = pd.read_csv('/content/drive/MyDrive/data/data.csv')

In [7]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [10]:
data = np.array(df['data'])

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [11]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

Оскільки кількість значень більше 30 - використовую z-test і стандартне значення статистичної значущості 5%. Нульова гіпотеза: середній чек за липень 2022 року не відрізняється від середнього чеку до цього місяця (не змінився). Альтернативна гіпотеза: середній чек за липень 2022 року більший за середній чек до цього місяця.

In [18]:
null_mean = 20
alpha = 0.05
ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')

In [16]:
ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

In [19]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


Відповідно до результатів проведення тесту бачимо, що нульова гіпотеза є правдивою, тобто середній чек статистично значуще не змінився. Можна також перевірити це знайшовши середню суму чеку:

In [20]:
np.mean(data)

19.37755319670441

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [21]:
df2 = pd.read_csv('/content/drive/MyDrive/data/data_2.csv')

In [22]:
data2 = np.array(df2['data'])

In [23]:
data2.shape

(120,)

In [24]:
np.mean(data2)

19.37322276512519

In [25]:
ztest(data, data2, value=0)

(0.005246673456808504, 0.9958137794593356)

In [27]:
ztest(data, data2, alternative='two-sided')

(0.005246673456808504, 0.9958137794593356)

Статистика для двох вибірок z-критерію становить 0,005, а відповідне p-value дорівнює 0.996

Оскільки p-value > 0.05, у нас немає підстав, щоб відкинути нульову гіпотезу. Це означає, що середні значення в липні та серпні 2022 року - однакові. Такий висновок також підтверджується знайденими середніми значеннями в кожному наборі даних.